main.tf

```diff
@@ -0,0 +1,377 @@
+  count = local.is_linux ? 1 : 0
+
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = merge(var.tags, var.tags_virtual_machine)
+
+  admin_password                                         = local.admin_password
+  admin_username                                         = var.admin_username
+  allow_extension_operations                             = var.allow_extension_operations
+  availability_set_id                                    = var.availability_set_id
+  bypass_platform_safety_checks_on_user_schedule_enabled = var.bypass_platform_safety_checks_on_user_schedule_enabled
+  computer_name                                          = coalesce(var.computer_name, split("-", trimprefix(var.name, "vm-"))[0])
+  custom_data                                            = var.custom_data
+  disable_password_authentication                        = false
+  encryption_at_host_enabled                             = var.encryption_at_host_enabled
+  license_type                                           = var.license_type
+  network_interface_ids                                  = local.network_interface_ids
+  patch_assessment_mode                                  = var.patch_assessment_mode
+  patch_mode                                             = var.patch_mode
+  provision_vm_agent                                     = var.provision_vm_agent
+  proximity_placement_group_id                           = var.proximity_placement_group_id
+  secure_boot_enabled                                    = var.secure_boot_enabled
+  size                                                   = var.size
+  source_image_id                                        = var.source_image_id
+  virtual_machine_scale_set_id                           = var.virtual_machine_scale_set_id
+  vtpm_enabled                                           = var.vtpm_enabled
+  zone                                                   = var.zone
+
+  dynamic "additional_capabilities" {
+    for_each = var.additional_capabilities[*]
+    content {
+      ultra_ssd_enabled   = var.additional_capabilities.ultra_ssd_enabled
+      hibernation_enabled = var.additional_capabilities.hibernation_enabled
+    }
+  }
+
+  dynamic "admin_ssh_key" {
+    for_each = var.authentication_type == "SSH" ? [true] : []
+    content {
+      username   = var.admin_username
+      public_key = local.admin_ssh_public_key
+    }
+  }
+
+  dynamic "boot_diagnostics" {
+    for_each = var.boot_diagnostics.enable == true ? [true] : []
+    content {
+      storage_account_uri = var.boot_diagnostics.storage_account_uri
+    }
+  }
+
+  dynamic "identity" {
+    for_each = try(var.identity.type, null) != null ? [true] : []
+    content {
+      type         = var.identity.type
+      identity_ids = try(coalescelist(var.identity.identity_ids, azurerm_user_assigned_identity.this[*].id), null)
```

Comment:
Remove `try()`?

Reply:
`var.identity` may be `null` (default). `var.identity.identity_ids` will fail in this case, and `try()` is therefore mandatory here.